cf-execd/cf-execd.c

```diff
@@ -602,7 +602,18 @@ static int SetupRunagentSocket(const ExecdConfig *execd_config)
     {
         sock_info.sun_family = AF_LOCAL;
 
-        MakeParentDirectory(sock_info.sun_path, true, NULL);
+        bool created;
+        MakeParentDirectory(sock_info.sun_path, true, &created);
+
+        /* Make sure the permissions are correct if the directory was created
+         * (note: this code doesn't run on Windows). */
+        if (created)
+        {
+            char *last_slash = strrchr(sock_info.sun_path, '/');
+            *last_slash = '\0';
+            chmod(sock_info.sun_path, (mode_t) 0700);
+            *last_slash = '/';
+        }
```

Comment:
Shouldn't this be done while creating the directory? By setting umask, creating, and then restoring?

Reply:
Yes, it should, but that's what `MakeParentDirectory()` is already doing. So I'd have to change it or reimplement it.

Reply:
Makes sense 👍